Sources/Datasets/Dataset.swift

```diff
@@ -1,7 +1,10 @@
 import Foundation
 
 protocol S5TFDataLoader: Sequence, IteratorProtocol {
-    func batched(_ batchSize: UInt) -> Self
+    var batchSize: Int { get }
+    var split: S5TFSplit { get }
+    func batched(_ batchSize: Int) -> Self
+    init(split: S5TFSplit, batchSize: Int)
```

Comment:
I'm not sure if `split` should be a property on `S5TFDataLoader` because that would restrict us from using it in other ways in the future.

For example, we could create a `flowFromDirectory` or `createFromCSV` at some point. These have undermined splits.

I think it would be best to make that a property on an `MNISTDataLoader`.

Reply:
We can default to the train split, perhaps, like TFDS does.